docs/insomnia/introduction-to-plugins.md

```diff
@@ -91,24 +106,85 @@ Example: Plugin package.json file
+      
+      // Modify
+      resp.__randomNumber = Math.random();
+
+      // If you want something from a user, use a prompt:
+      const promptResult = await ctx.app.prompt('Type something', { defaultValue: 'abcd' });
+      resp.__customValue = promptResult;
+
+      ctx.response.setBody(jsonObjToBuffer(resp));
+    } catch {
+      // no-op
+    }
+  }
+]
+```
+_This example adds a `__randomNumber` and `__customValue` property to a JSON response. Update the functionality as needed._
+
 ## Manage Plugins
-Plugins can be added from the Plugins tab within the application preferences dialog. They can also be downloaded and installed directly from npm.
+
+The **Plugins** tab in the **Preferences** menu enables the following functionality:
+
+* Add an existing plugin
+* Create a new plugin locally
+* Toggle a plugin to enable or disable it
+* Reveal the exact local plugin location on your machine
+
+{:.alert .alert-primary}
+**Note**: Plugins can also be downloaded and installed directly from [NPM](https://insomnia.rest/plugins).
 
 ## Publish Plugins
-Before you publish your plugin, there are two main things you want to ensure if you want your plugin to be recognized by Insomnia and be available through the Insomnia Plugin Hub.
 
-* Your plugin's package.json contains the insomnia attribute with the correct structure. See Create a Plugin for more info.
-* Your plugin's package name is prefixed with `insomnia-plugin-`
+Before you publish your plugin, ensure you have met the following criteria for your plugin to be recognized by Insomnia and be available on the [Insomnia Plugin Hub](https://insomnia.rest/plugins).
+
+Your plugin should: 
+
+* Include a correctly structured `package.json` file containing the `insomnia` attribute. See [Plugin package.json](#plugin-packagejson) for more info.
+* Have a package name prefixed with `insomnia-plugin-`.
+* Be publicly available. 
+
+After you have verified that your plugin meets the criteria described above, publish your public plugin following the [NPM publish unscoped public packages instructions](https://docs.npmjs.com/creating-and-publishing-unscoped-public-packages). Publish an _unscoped_ package to ensure it appears on the Insomnia Plugin Hub. 
+
+If your package does not show up on the Insomnia Plugin Hub after a few days, please [contact us](https://insomnia.rest/support) with the name of your plugin and a link to the published NPM package.
+
+### Publish Scoped Plugins
+
+Insomnia can also use private (scoped) plugins. This commonly enables enterprise users to keep the plugin private from Insomnia. 
+
+To enable private (scoped) plugins:
 
-After you have verified that your plugin meets the criteria described above, now you can proceed with publishing your plugin following the npm instructions on publishing a package.
+1. Navigate to the `plugins` directory in [Application Data](/insomnia/application-data).
+2. Run `npm install @scoped/plugin`.
```

Comment:
Is there a step needed her to open in terminal? Not sure what level of detail is necessary.

In any case, here's a screenshot showing each case working. The first one was just put directly into the plugins folder, the 2nd and 3rd can be installed from within Insomnia, and only the 3rd will show on the Plugin Hub.

<img width="314" alt="image" src="https://user-images.githubusercontent.com/4312346/133012282-c2fda43d-e731-4d0d-a624-4180c395d185.png">

Reply:
Thank you for the screenshot! No need to get that granular given the audience.